lib/origen/application/lsf.rb

```diff
@@ -22,9 +23,31 @@ class Configuration
         def initialize
           @group = nil
           @project = 'msg.te'
-          @resource = 'linux'
-          @queue = 'short'
           @debug = false
+
+          # The following will grep the output of the bqueues command. It will search for the queue based on
+          # the LSF configuration and assign the queue accordingly. This assumes that the queues are named short or shortq
+          # and probably can be modified to parse out whatever we need as the queue name.
+          # The resource is currently assigned as either linux or rhel6 as well dependent on the queue assignment
+          `bqueues > /tmp/queuenames.log`
+          file = File.open('/tmp/queuenames.log')
+          columns = []
+          file.each_line do |line|
+            columns << line.split(' ')[0, 11]
+          end
+          queuenames = []
+          # rubocop:disable Style/For: Prefer each over for
+          for i in 1..(columns.size - 1)
```

Comment:
Should listen to rubocop here, should be:

~~~ruby
(1..(columns.size - 1)).each do |i|
  #...
~~~